### PR TITLE
refactor(meta): use `MetadataModelError` instead of `RwError` in trait `MetadataModel`

### PR DESCRIPTION
**Pull request by HuaHuaY**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Now there is a call path `Meta -> Hummock -> MetaStore` in meta when meta receives a RPC request about hummock. The types of return error are like `RwError <- Hummock's Error <- RwError`. Both meta and meta store use `RwError` as return value. 

Even though the circular reference doesn't prevent us from compiling, I add a new error type `MetadataModelError`, which breaks the circular reference. This pr is aimed to make the error organization clearer and eventually help us to replace the `RwError` with meta's own error type.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#4168

### CODE REVIEW

File: src/meta/src/model/error.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::error::{ErrorCode, RwError};
+use risingwave_pb::ProstFieldNotFound;
+use thiserror::Error;
+
+use crate::storage::Error as MetaStoreError;
```
Comment (BugenZhao) [inline src/meta/src/model/error.rs:19]: How about just renaming this struct to `MetaStoreError`?

File: src/meta/src/model/error.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+impl From<ProstFieldNotFound> for MetadataModelError {
+    fn from(p: ProstFieldNotFound) -> Self {
+        MetadataModelError::InternalError(anyhow::anyhow!(
+            "Failed to decode prost: field not found `{}`",
+            p.0
+        ))
+    }
+}
+
+impl From<MetadataModelError> for RwError {
```
Comment (BugenZhao) [inline src/meta/src/model/error.rs:44]: May add a TODO here for removing this later.

File: src/meta/src/hummock/error.rs [hunk 3]
```diff
@@ -99,15 +100,11 @@ impl From<Error> for risingwave_common::error::RwError {
...
-        match error.inner() {
-            ErrorCode::InternalError(err) => Error::InternalError(err.to_owned()),
-            ErrorCode::ItemNotFound(err) => Error::InternalError(err.to_owned()),
-            _ => {
-                panic!("conversion not supported");
-            }
+impl From<MetadataModelError> for Error {
+    fn from(err: MetadataModelError) -> Self {
+        match err {
+            MetadataModelError::MetaStoreError(e) => e.into(),
```
Comment (BugenZhao) [inline src/meta/src/hummock/error.rs:106]: Will hummock manager directly yield `MetaStoreError`? If not, I suggest replacing it with `MetadataModelError` and directly wrap an error in that variant.

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM. Good work!

**Review by yezizp2012 — APPROVED** [review]
Good job, generally LGTM.

File: src/meta/src/hummock/error.rs (hunk 3 above)
Comment (HuaHuaY) [reply src/meta/src/hummock/error.rs:106]: Let us do it later when we replace `RwError` to meta's own error.

**Pull request merged**